### PR TITLE
tools: added option to change how duckduckgosearchresults tool converts api outputs into string

### PR DESCRIPTION
**Pull request by isahers1**

### CODE REVIEW

File: libs/community/langchain_community/tools/ddg_search/tool.py [hunk 1]
```diff
@@ -54,16 +54,19 @@ class DuckDuckGoSearchResults(BaseTool):
...
     def _run(
         self,
         query: str,
         run_manager: Optional[CallbackManagerForToolRun] = None,
     ) -> str:
         """Use the tool."""
-        res = self.api_wrapper.results(query, self.max_results, source=self.backend)
-        res_strs = [", ".join([f"{k}: {v}" for k, v in d.items()]) for d in res]
-        return ", ".join([f"[{rs}]" for rs in res_strs])
+        res = self.default_api_wrapper_to_str(self.api_wrapper.results(query, self.max_results, source=self.backend))
```
Comment (baskaryan) [inline libs/community/langchain_community/tools/ddg_search/tool.py:66]: delete self.default_api_wrapper_to_str

File: libs/community/langchain_community/tools/ddg_search/tool.py [hunk 2]
```diff
@@ -54,16 +54,19 @@ class DuckDuckGoSearchResults(BaseTool):
     )
     backend: str = "text"
     args_schema: Type[BaseModel] = DDGInput
+    keys_to_include: Optional[List[str]] = None
+    seperator_between_results: str = ", "
```
Comment (baskaryan) [inline libs/community/langchain_community/tools/ddg_search/tool.py:58]: `results_separator` seems a bit more succinct

File: libs/community/langchain_community/tools/ddg_search/tool.py [hunk 3]
```diff
@@ -62,8 +64,9 @@ def _run(
     ) -> str:
         """Use the tool."""
         res = self.api_wrapper.results(query, self.max_results, source=self.backend)
-        res_strs = [", ".join([f"{k}: {v}" for k, v in d.items()]) for d in res]
-        return ", ".join([f"[{rs}]" for rs in res_strs])
+        res_strs = [", ".join([f"{k}: {v}" for k, v in d.items() 
+                    if not self.keys_to_include or k in self.keys_to_include]) for d in res]
+        return self.results_separator.join([f"[{rs}]" for rs in res_strs])
```
Comment (baskaryan) [inline libs/community/langchain_community/tools/ddg_search/tool.py:69]: should we replace `f"[{rs}]"` with just `rs`? not obviouus to me former should be hardcoded, especially if separator is configurable

Comment (isahers1) [reply libs/community/langchain_community/tools/ddg_search/tool.py:69]: Can I just replace with .join(res_strs) then?

File: libs/community/tests/integration_tests/document_loaders/test_sitemap.py
Comment (baskaryan) [inline libs/community/tests/integration_tests/document_loaders/test_sitemap.py:1]: assume this wasn't meant to be part of this pr?

**Pull request merged**